lib/rubocop/cop/betterment/status_code.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Betterment
+      class StatusCode < Base
+        extend AutoCorrector
+
+        ACTIONS = %i(create update destroy).freeze
+
+        REDIRECT_MSG = <<~MSG.gsub(/\s+/, " ")
+          Did you forget to specify an HTTP status code? The default is `status: :found`, which
+          is usually inappropriate in this situation. Use `status: :see_other` when redirecting a
+          POST, PUT, PATCH, or DELETE request to a GET resource.
+        MSG
+
+        RENDER_MSG = <<~MSG.gsub(/\s+/, " ")
+          Did you forget to specify an HTTP status code? The default is `status: :ok`, which might
+          be inappropriate in this situation. Rendering after a POST, PUT, PATCH or DELETE request
+          typically represents an error (e.g. `status: :unprocessable_entity`).
+        MSG
```

Comment:
I don't have super strong feelings about this one, but I do think that there are enough cases where I'd like to render a `status: :ok` from a form submission without having to disable a linter.

But I guess, important question: are there known downsides to rendering a 200 instead? (Changes in browser caching behavior, etc?)

Assuming no meaningful behavioral differences, I'd be inclined to ship only the `status: :see_other` enforcement first and then decide if we want to separately enforce `status: :unprocessable_entity` on re-renders.

But again, I don't feel too strongly. Perhaps there's a slightly more complex enforcement rule that could check if your `render` sits within an `else` after a `.save`. I think I'd be more confident in a blanket enforcement rule that actually checked that an entity processing attempt was attempted and that the `render `is in the "did not successfully process" branch.

Reply:
Maybe I just need to clarify the error message here. I'm really just checking that you've explicitly provided a status. If you want to render with `status: :ok`, that's totally find and valid.

To disregard this linter warning, you'd just add `status: :ok`. You wouldn't need to disable the linter.

Reply:
The downside of rendering a 200 is this:
> When a web form is submitted to a server through an [HTTP](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol) [POST](https://en.wikipedia.org/wiki/POST_(HTTP)) request, attempts to refresh the server response can cause the contents of the original POST to be resubmitted, possibly causing undesired results, such as a duplicate web [purchase](https://en.wikipedia.org/wiki/Purchasing).[[1]](https://en.wikipedia.org/wiki/Post/Redirect/Get#cite_note-1) Some browsers mitigate this risk by warning the user that they are about to re-issue a POST request.
> -- https://en.wikipedia.org/wiki/Post/Redirect/Get#:~:text=Post%2FRedirect%2FGet%20(PRG,problem%20encountered%20in%20user%20agents.

I suspect that in many of these cases, it might be better to redirect to a success page. I don't feel confident saying you should _never_ return a 200, but I do think it's reasonable to say "hey developer, are you sure you want to render a 200 here? If so, just add `status: :ok` to that render call".